### PR TITLE
HPCC-31294 Refactor the way job ids are passed to the logging

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/jlib/jlog.cpp [hunk 1]
```diff
@@ -1289,15 +1285,15 @@ void LogMsgPrepender::report(const LogMsgCategory & cat, const char * format, ..
     buff.append(file).append("(").append(line).append(") : ").append(format);
     va_list args;
     va_start(args, format);
-    queryLogMsgManager()->report_va(cat, unknownJob, buff.str(), args);
+    queryLogMsgManager()->report_va(cat,  buff.str(), args);
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:1288]: trivial/formatting: double space

File: system/jlib/jlog.cpp [hunk 2]
```diff
@@ -1289,15 +1285,15 @@ void LogMsgPrepender::report(const LogMsgCategory & cat, const char * format, ..
...
     va_end(args);
 }
 
 void LogMsgPrepender::report_va(const LogMsgCategory & cat, const char * format, va_list args)
 {
     StringBuffer buff;
     buff.append(file).append("(").append(line).append(") : ").append(format);
-    queryLogMsgManager()->report_va(cat, unknownJob, buff.str(), args);
+    queryLogMsgManager()->report_va(cat,  buff.str(), args);
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:1296]: trivial/formatting: double space

File: system/jlib/jlog.cpp [hunk 3]
```diff
@@ -1323,55 +1319,12 @@ void LogMsgPrepender::report(const LogMsgCategory & cat, const IException * exce
...
-}
-
-void LogMsgPrepender::report(const LogMsgCategory & cat, const LogMsgJobInfo & job, const IException * exception, const char * prefix)
-{
-    StringBuffer txt;
-    if (prefix) 
-        txt.append(prefix).append(" : ");
-    exception->errorMessage(txt);
-    queryLogMsgManager()->report(cat, job, exception->errorCode(), "%s(%d) : %s", file, line, txt.str());
+    queryLogMsgManager()->report(cat,  exception->errorCode(), "%s", buff.str());
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:1322]: trivial/formatting: double space

..and a number of others in changes below.

File: system/jlib/jlog.cpp [hunk 4]
```diff
@@ -3430,65 +3339,31 @@ void ctxlogReport(const LogMsgCategory & cat, const char * format, ...)
...
     va_list args;
     va_start(args, format);
-    ctxlogReportVA(cat, unknownJob, code, format, args); 
+    ctxlogReportVA(cat,  code, format, args);
     va_end(args);
 }
 void ctxlogReportVA(const LogMsgCategory & cat, LogMsgCode code, const char * format, va_list args)
 {
-    ctxlogReportVA(cat, unknownJob, code, format, args); 
+    ctxlogReportVA(cat,  code, format, args);
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:3359]: This is the recursive call.
Should it be:
```
    queryLogMsgManager()->report_va(cat, code, format, args);
```

File: system/jlib/jlog.cpp [hunk 5]
```diff
@@ -3430,65 +3339,31 @@ void ctxlogReport(const LogMsgCategory & cat, const char * format, ...)
...
-    }
-    else
-        queryLogMsgManager()->report_va(cat, job, code, format, args);
-}
-void ctxlogReport(const LogMsgCategory & cat, const LogMsgJobInfo & job, const IException * e, const char * prefix)
-{
-    StringBuffer buff;
-    e->errorMessage(buff);
-    ctxlogReport(cat, job, e->errorCode(), "%s%s%s", prefix ? prefix : "", prefix ? prefix : " : ", buff.str());
+    ctxlogReport(cat,  e, prefix);
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:3363]: also recursive I think. Should it be:
```
    queryLogMsgManager()->report(cat, e, prefix);
``` @

**Comment by jakesmith** [comment]
@ghalliday - there are some build issues, e.g. - https://github.com/hpcc-systems/HPCC-Platform/actions/runs/7928428731/job/21646656101?pr=18304

**Review by jakesmith — COMMENTED** [review]
@ghalliday - looks eminently sensible to me.

A couple of recursive definitions, and some spurious spaces introduced.

File: system/jlib/jlog.cpp (hunk 1 above)
Comment (ghalliday) [reply system/jlib/jlog.cpp:1288]: Yes, an unfortunate side-effect of a global search and replace that didn't include the trailing space.  I'll try and fix.

**Comment by ghalliday** [comment]
@jakesmith I have now rebased this, squashed the previous commits, and added a couple more commits to clean the code up further

File: thorlcr/master/thgraphmanager.cpp [hunk 6]
```diff
@@ -1469,10 +1465,9 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
                             {
                                 if (!streq(currentWuid, wuid))
                                 {
-                                    queryLogMsgManager()->removeJobId(thorJob.queryJobID());
-                                    LogMsgJobId thorJobId = queryLogMsgManager()->addJobId(wuid);
-                                    thorJob.setJobID(thorJobId);
-                                    setDefaultJobId(thorJobId);
+                                    activeJobName.clear();
+                                    activeJobName.set(wuid);
```
Comment (jakesmith) [inline thorlcr/master/thgraphmanager.cpp:1469]: trivial: it's fine as it is, looking at these 2 lines, it made me think 'set' should implicitly clear if it needs to.

**Review by jakesmith — APPROVED** [review]
@ghalliday - I can't spot any issues (1 trivial comment).

**Comment by richardkchapman** [comment]
I did not spot any issues

File: thorlcr/master/thgraphmanager.cpp (hunk 6 above)
Comment (ghalliday) [reply thorlcr/master/thgraphmanager.cpp:1468]: changed

**Pull request merged**